include/mlir/Dialect/StandardOps/Ops.td

```diff
@@ -950,6 +986,15 @@ def SelectOp : Std_Op<"select", [NoSideEffect, SameOperandsAndResultShape]> {
   let hasFolder = 1;
 }
 
+def SignFOp : FloatUnaryOp<"signf"> {
+  let summary = "floating point sign operation";
+  let description = [{
+    The `signf` operation takes one operand and returns one result of the same type.
+    This type may be a float scalar type, a vector whose element type is float, or a
+    tensor of floats. It has no standard attributes.
+  }]
+}
+
```

Comment:
Where are "signf" and "negf" coming from? What motivates adding these to the standard dialect?
(for instance I don't think LLVM has these).

Reply:
They will be lowered to copysign when we do LLVM legalization.

We can replace these ops with the use of `copysign`. Then the lowering from LHLO will be a bit longer, but STD->LLVM will be trivial. What do you prefer? Is the point of Standard dialect to be as close to LLVM, but still allow to lower to LLVM, Cuda, Rocm if needed?

Reply:
> Is the point of Standard dialect to be as close to LLVM, but still allow to lower to LLVM, Cuda, Rocm if needed?

Standard dialect does not really have a well defined purpose, but it is central enough that we should have a higher level of scrutiny and consideration about what is going there.

Can you send an email to mlir@tensorflow.org to propose these, explaining their relationship to `copysign`?